lib/SQL/Abstract/mysql.pm

```diff
@@ -0,0 +1,263 @@
+package SQL::Abstract::mysql;
+use Mojo::Base 'SQL::Abstract';
+
+BEGIN { *puke = \&SQL::Abstract::puke }
+
+sub insert {
+  my $self    = shift;
+  my $table   = $self->_table(shift);
+  my $data    = shift || return;
+  my $options = shift || {};
+
+  my $method = $self->_METHOD_FOR_refkind('_insert', $data);
+  my ($sql, @bind) = $self->$method($data);
+  my $command;
+
+  # options
+  if (exists $options->{on_conflict}) {
+    my $on_conflict = $options->{on_conflict} // '';
+    my %commands = (ignore => 'insert ignore', replace => 'replace');
+    if (ref $on_conflict eq 'HASH') {
+      $command = 'insert';
+      my ($sql2, @bind2) = $self->_update_set_values($on_conflict);
+      $sql .= $self->_sqlcase(' on duplicate key update ') . $sql2;
+      push @bind, @bind2;
+    }
+    else {
+      $command = $commands{$on_conflict} or puke qq{on_conflict value "$on_conflict" is not allowed};
+    }
+  }
+  else {
+    $command = 'insert';
+  }
+  $sql = join ' ', $self->_sqlcase("$command into"), $table, $sql;
+
+  return wantarray ? ($sql, @bind) : $sql;
+}
+
+sub new {
+  return shift->SUPER::new(quote_char => chr(96), name_sep => '.');
```

Comment:
Why did you move this here instead of keeping it in Mojo::mysql? Shouldn't it be possible to customise this?

Reply:
We could leave it the same way as Mojo::Pg does, but then we should document it.

Reply:
I would appreciate that. Or at least pass in `@_` so the end user can override the settings.

Reply:
I'll move it back and update the docs.